packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -198,54 +197,33 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin {
 				source: JSON.stringify(config),
 			});
 		},
+		handleHotUpdate(options) {
+			if (resolvedPluginConfig.configPaths.has(options.file)) {
+				options.server.restart();
+			}
+		},
+		async buildEnd() {
+			if (miniflare) {
+				await miniflare.dispose();
+				miniflare = undefined;
+			}
+		},
 		async configureServer(viteDevServer) {
-			let error: unknown;
-
-			const miniflare = new Miniflare(
+			miniflare = new Miniflare(
 				getDevMiniflareOptions(resolvedPluginConfig, viteDevServer),
 			);
 
 			await initRunners(resolvedPluginConfig, viteDevServer, miniflare);
-
-			viteDevServer.watcher.on('all', async (_, path) => {
```

Comment:
ah ok, so we were already restarting on wrangler config changes! 🤔

Reply:
Yes, but we were restarting Miniflare but not Vite. Now that `name`, `main` etc. are read from the Wrangler config file, that's not enough.

Reply:
ah yeah yeah I remember now 🙂👍 

(anyways it's nice to see that the functionality we want is all there but we get simpler code ❤️)